docs/index.rst

```diff
@@ -0,0 +1,50 @@
+Jupyter Kernel Management |version|
+===================================
+
+This package provides the Python API for starting, managing and communicating with
+Jupyter kernels.
+For information on messaging with Jupyter kernels, please
+refer to the `Jupyter Protocol documentation (FIXME) <https://jupyter-client.readthedocs.io/en/latest/index.html>`_.
+
+.. note::
+   This is a new interface under development, and may still change.
+   Not all Jupyter applications use this yet.
+   See the `jupyter_client <https://pexpect.readthedocs.io/en/latest/>`_ docs for the established way of
+   discovering and managing kernels.
+
+
+.. toctree::
+   :maxdepth: 2
+   :caption: User Documentation
+
+   kernel_finder
+   kernel_manager
+   kernel_client
+   kernel_restarter
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Developer documentation
```

Comment:
I don't think this is 'users' vs 'developers' - for a library, users are developers. I see the distinction you're getting at, though - maybe 'application developers' vs 'kernel provider developers'. Not exactly snappy, though.

Reply:
Actually, this was copied from protocol (and thereby client). :smile:  I like the alternate sections better.  I agree that 'kernel provider developers' is a bit long-winded, but I think having 'kernel' in the section header like is important.  (Provider developers seems fine, but that additional explicitness is probably warranted at that level of the docs)